packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -1135,10 +1158,22 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
         if (shouldRetryRequest) {
             this.stats.errorTrackerRetry.add(error);
 
-            await this._tagUserHandlerError(() => this.errorHandler?.(this._augmentContextWithDeprecatedError(crawlingContext, error), error));
+            if (error instanceof SessionError) {
```

Comment:
this whole block is pretty complex, lets extract it to a private method and try to refactor to remove some nesting ideally